### ISSUE TITLE
Chart rendering markers when it shouldn't (styledMode)

### ISSUE THREAD

**Issue by kkesidis** [highcharts/highcharts#23329]
#### Expected behaviour

The chart should not render the series markers until we hover over each point.

#### Actual behaviour

The chart displays all markers, when it shouldn't 

#### Live demo with steps to reproduce
https://jsfiddle.net/xkvznp3c/42/

To trigger the bug switch a few times between the "Set extremes" custom button and a range button from the chart (i was able to trigger it with 1y and All). After a few times, the series will always render their markers at all times.

#### Product version
v12.3.0

#### Affected browser(s)
All

**Comment by kkesidis** [comment]
should have added from the start, but in our implementation the markers are not always visible (when they shouldn't), they don't follow the line direction either

<img width="2560" height="723" alt="Image" src="https://github.com/user-attachments/assets/cb6136b8-5f97-4d90-8c36-db64cbe32f33" />

**Comment by bre1470** [comment]
Hello @kkesidis ,

Thank you for the report. Unfortunately I am not able to reproduce it on PC and smartphone.
Therefor some questions:

1. On which platform(s) and browser(s) does this happen?
2. Does it occurs with a pointer device (mouse etc.) and/or touch screen (finger etc)?
3. Do you have to hover over the series before switching the range?

**Comment by kkesidis** [comment]
Hello @bre1470,

I can reproduce this by clicking between the "Set extremes" (custom button) and the 1Y chart button a few times and then this happens

<img width="638" height="303" alt="Image" src="https://github.com/user-attachments/assets/0e1e6dd4-3621-4778-855c-bc362c7001d2" />

I am using a Mac M1 Pro and i can reproduce the issue on both Chrome and Firefox. Was also able to reproduce on Windows. Something i didn't think to check before is that this issue is triggered only after a certain chart width. I can trigger it for a chart with width > 1500px (or zoomed out enough on smaller screens).

**Comment by bre1470** [comment]
Thank you for the quick response, @kkesidis!

Indeed with a larger chart size, the markers do appear after "1Y" activation.
This happens since Highcharts v10, so if it is a critical issue then Highcharts v9.3.3 could be used to avoid it.
As this is quite outdated, I created a workaround while we look into a bugfix: https://jsfiddle.net/bre1470/b71uhrqj/

```TypeScript
        chart: {
            events: {
                redraw: () => {
                    this.series.forEach(series => {
                        if (!series.group.attr('class').includes('highcharts-a11y-markers-hidden')) {
                            series.group.addClass('highcharts-a11y-markers-hidden');
                        }
                    });
                }
            },
// ...
```

Let me know, in case it does not help.

Internal note: This is caused by the accessibility module as the `highcharts-a11y-markers-hidden` class on series groups disappears after using one of the defined ranges.

**Comment by kkesidis** [comment]
Hello @bre1470, i can still reproduce this with the above script in the `chart.events`. https://jsfiddle.net/L2rbmho9/

**Comment by bre1470** [comment]
Hello @kkesidis !
Thank you for the feedback. Make sure to use `redraw: () => {` pattern. See: https://jsfiddle.net/bre1470/wsv57xgL/

**Comment by kkesidis** [comment]
But, if i use the arrow function pattern (like the example you provided), i get console errors

<img width="679" height="292" alt="Image" src="https://github.com/user-attachments/assets/0935dabd-e72f-4557-8f24-012467b3162f" />

even when i `console.log(this)` the object i see does not have a series property

**Comment by bre1470** [comment]
Thank you for pointing this out.
Then the workaround is not acceptable and just prevents the problem by throwing an error.
The problem seems to be a bit more complicated than expected and issue priority has been increased.

**Comment by bre1470** [comment]
Internal: bisect to https://github.com/highcharts/highcharts/commit/99479ccbcedd02ee7e67853877535cd873a71f12
